pygdf/categorical.py

```diff
@@ -47,11 +46,14 @@ def __init__(self, dtype, codes_dtype, categories, ordered):
         self._codes_impl = numerical.NumericalSeriesImpl(codes_dtype)
 
     def __eq__(self, other):
-        if isinstance(other, CategoricalSeriesImpl):
-            return all([self.dtype == other.dtype,
-                        tuple(self._categories) == tuple(other._categories),
-                        self._ordered == other._ordered,
-                        self._codes_impl == other._codes_impl])
+        try:
+            return (isinstance(other, CategoricalSeriesImpl) and
+                    self.dtype == other.dtype and
+                    tuple(self._categories) == tuple(other._categories) and
+                    self._ordered == other._ordered and
+                    self._codes_impl == other._codes_impl)
+        except Exception:
+            return False
```

Comment:
The `except` will just hide accidental code error.  If `other` is a `NumericalSeriesImpl`, it will just return `None`, which will be interpreted as `False`.  I think the try-except should be removed.